src/test/cypress/support/requests/CourseManagementRequests.ts

```diff
@@ -258,6 +281,39 @@ export class CourseManagementRequests {
         });
     }
 
+    startExerciseParticipation(courseId: number, exerciseId: number) {
+        return cy.request({
+            url: `${COURSE_BASE}${courseId}/exercises/${exerciseId}/participations`,
+            method: POST,
+        });
+    }
+
+    makeTextExerciseSubmission(exerciseId: number, text: string) {
+        return cy.request({
+            url: `${EXERCISE_BASE}${exerciseId}/text-submissions`,
+            method: PUT,
+            body: { submissionExerciseType: 'text', text, id: null },
+        });
+    }
+
+    updateTextExerciseDueDate(exercise: any, due = day()) {
+        exercise.dueDate = dayjsToString(due);
+        return this.updateTextExercise(exercise);
+    }
+
+    updateTextExerciseAssessmentDueDate(exercise: any, due = day()) {
+        exercise.assessmentDueDate = dayjsToString(due);
+        return this.updateTextExercise(exercise);
+    }
```

Comment:
Could you not expose the updateTextExercise(exercise: any) function and simply edit the exercise object in the test? What if i want to change something else in a text exercise do i just make another function that changes just one particular field? Seems odd to me.

Reply:
Yeah I could expose it. The idea behind keeping it private and adding those functions was to separate the logic, so that the test does not have to know the internal structure of the exercise DTO. So in the future if the "assessmentDueDate" field was renamed we would only have to change it in this one function instead of every test, which uses that function.
But at the moment this doesn't provide too much benefit because the functions are only called in one test once anyways.